### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by TheRealNoob**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    apt install git curl -y && \
+    apt clean autoclean && \
+    apt autoremove --yes && \
+    rm -rf /var/lib/{apt,dpkg,cache,log}
+
+RUN pip install --no-cache-dir -r /root/requirements.txt
+
+COPY . /root/
+
+CMD uvicorn api.app:app --proxy-headers --host 0.0.0.0 --port ${port} --root-path ${path}
```
Comment (ThorntonMatthewD) [inline Dockerfile:23]: Is not using the [] syntax poor practice? I can't seem to get env vars to insert into that setup.

Comment (TheRealNoob) [reply Dockerfile:23]: I could be wrong on this, but I think I had it set up the way you're talking about.  I think this commit broke it.
https://github.com/TheRealNoob/Bot-Detector-Core-Files/commit/7ba82a77f8681bc3468c3d19c6dabd09ecde78f0

Comment (TheRealNoob) [reply Dockerfile]: fixed with https://github.com/TheRealNoob/Bot-Detector-Core-Files/commit/52556b4a751bc3db94a5510b749c65bc2a3edecd

**Comment by extreme4all** [comment]
i can't seem to access http://localhost:5000/docs when using docker-compose

**Comment by extreme4all** [comment]
deduced it to be a local issue

**Comment by extreme4all** [comment]
image
image

**Comment by extreme4all** [comment]
ignore the error

**Comment by ThorntonMatthewD** [comment]
@extreme4all Does your Dockerfile look something like this?

```
FROM python:3.10-slim

ARG api_port
ENV port ${api_port}

ARG root_path
ENV path ${root_path}

WORKDIR /root
COPY ./requirements.txt /root/requirements.txt

# required to install github.com/TheRealNoob/python-logging-loki.git
RUN apt update && \
    apt install git curl -y && \
    apt clean autoclean && \
    apt autoremove --yes && \
    rm -rf /var/lib/{apt,dpkg,cache,log}

RUN pip install --no-cache-dir -r /root/requirements.txt

COPY . /root/

CMD ["uvicorn", "api.app:app", "--proxy-headers", "--host", "0.0.0.0", "--port", " $port", "--root-path", " $path"]
```

**Comment by ThorntonMatthewD** [comment]
Right now the docker-compose file doesn't provide the variables. This will fix that:
```
  api:
    build:
      context: .
      dockerfile: Dockerfile
      args:
          root_path: "/"
          api_port: "5000"
    image: bot-detector/bd-api:latest
    ```

**Comment by extreme4all** [comment]
> @extreme4all Does your Dockerfile look something like this?
> 
> ```
> FROM python:3.10-slim
> 
> ARG api_port
> ENV port ${api_port}
> 
> ARG root_path
> ENV path ${root_path}
> 
> WORKDIR /root
> COPY ./requirements.txt /root/requirements.txt
> 
> # required to install github.com/TheRealNoob/python-logging-loki.git
> RUN apt update && \
>     apt install git curl -y && \
>     apt clean autoclean && \
>     apt autoremove --yes && \
>     rm -rf /var/lib/{apt,dpkg,cache,log}
> 
> RUN pip install --no-cache-dir -r /root/requirements.txt
> 
> COPY . /root/
> 
> CMD ["uvicorn", "api.app:app", "--proxy-headers", "--host", "0.0.0.0", "--port", " $port", "--root-path", " $path"]
> ```

yeah i think something went wrong opening the branch and i was on the existing branch not the PR

**Comment by extreme4all** [comment]
when running docker-compose build && docker-compose up the following error occurs 
image

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    apt install git curl -y && \
+    apt clean autoclean && \
+    apt autoremove --yes && \
+    rm -rf /var/lib/{apt,dpkg,cache,log}
+
+RUN pip install --no-cache-dir -r /root/requirements.txt
+
+COPY . /root/
+
+CMD ["uvicorn", "api.app:app", "--proxy-headers", "--host 0.0.0.0", "--port ${port}", "--root-path", "${path}"]
```
Comment (extreme4all) [inline Dockerfile:23]: i think it should be
`CMD ["uvicorn", "api.app:app", "--proxy-headers", "--host", "0.0.0.0", "--port", "${port}", "--root-path", "${path}"]`
like it was in the past
`CMD ["uvicorn", "api.app:app", "--proxy-headers", "--host", "0.0.0.0", "--port", "5000", "--root-path", "/dev"]`

**Review by extreme4all — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (ThorntonMatthewD) [reply Dockerfile:23]: I'm running into errors with the suggested format also. The env vars doesn't seem to be going in correctly..

This works: `CMD uvicorn api.app:app --proxy-headers --host 0.0.0.0 --port ${port} --root-path ${path}` but I don't think that is the syntax we want to use.

Comment (ThorntonMatthewD) [reply Dockerfile:23]: `Error: Invalid value for '--port': ' $port' is not a valid integer.`

Comment (ThorntonMatthewD) [reply Dockerfile:23]: Same issue with ${port} syntax as well.

Comment (ThorntonMatthewD) [reply Dockerfile:23]: @extreme4all @TheRealNoob I can't get anything except for `CMD uvicorn api.app:app --proxy-headers --host 0.0.0.0 --port ${port} --root-path ${path}` to work. https://github.com/moby/moby/issues/5509 is a good synopsis of why this is the case.

**Comment by extreme4all** [comment]
@TheRealNoob is it not possible to use ENV variables in docker?

**Comment by TheRealNoob** [comment]
> @TheRealNoob is it not possible to use ENV variables in docker?

It is.  [This is](https://docs.docker.com/engine/reference/builder/#environment-replacement) ENV variables in Dockerfiles.  [This page](https://docs.docker.com/compose/environment-variables/) talks about the different options you have for setting environment variables in compose.

File: docker-compose.yaml [hunk 3]
```diff
@@ -40,37 +42,37 @@ services:
...
-  #   depends_on:
-  #     - api
+  machine-learning:
+    build:
+      context: ../bot-detector-ML
+      dockerfile: Dockerfile
+    image: bot-detector/bd-ml:latest
+    environment:
+      - token=verify_ban
+      - api=http://api:6532/
```
Comment (extreme4all) [inline docker-compose.yaml:52]: isnt api at 5000?

**Comment by extreme4all** [comment]
This change is nearly good to go, just the mysql database needs some work

**Review by extreme4all — APPROVED** [review]
seems to build

**Pull request merged**